File: backend/src/main/kotlin/it/unicam/cs/pawm/model/Player.kt
package it.unicam.cs.pawm.model

import kotlinx.serialization.Serializable
import org.jetbrains.exposed.sql.Table

@Serializable
data class Player(
    val id: Int = -1,
    val name: String,
    val surname: String,
    val email: String,
    val password: String,
    val passwordSalt: String,
    //Aggiungere altri campi
)

object PlayerTable : Table() {
    val id = integer("id").autoIncrement()
    val name = varchar("name", 50)
    val surname = varchar("surname", 50)
    val email = varchar("email", 50).uniqueIndex()
    val password = char("password", 60)
    val passwordSalt = varchar("passwordSalt", 60)

    override val primaryKey = PrimaryKey(id)
}

File: frontend/src/model/Player.ts
import { Password } from "src/service/hash.service";

export class Player {
    id: number;
    name: string;
    surname: string;
    username: string;
    email: string;
    password: string;
    passwordSalt: string;

    constructor(id: number, name: string, surname: string, username: string, email: string, password: Password) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.username = username;
        this.email = email;
        this.password = password.passwordHash;
        this.passwordSalt = password.salt;
    }
}
